LoveMachine.Core/ButtplugController.cs

```diff
@@ -107,6 +110,31 @@ protected IEnumerator RunVibratorLoop(int girlIndex, Bone bone)
             }
         }
 
+        protected IEnumerator RunRotatorLoop(int girlIndex, Bone bone)
+        {
+            while (true)
+            {
+                if (IsIdle(girlIndex))
+                {
+                    yield return new WaitForSeconds(.1f);
+                    continue;
+                }
+                string pose = GetPose(girlIndex);

```

Comment:
Hmm, I don't think a rotator needs to wait for that since there's no top side

Reply:
I see, in that case my question is: What is this movement synced to? Isn't it basically just doing its own thing with little to no regard to what's happening on the screen? And if so, what is the benefit of having this logic inside the plugin and not in a separate script?

Reply:
> What is this movement synced to?

The movement synced to the speed of strokes.

Reply:
I don't want to micromanage things here, but I would imagine there is so much more that could be done with a rotator. Maybe rotate fast when the pp is going in, and a bit slower when it's going out? So that it feels like something is pulsating or idunno. That would be just two messages per stroke, any rotator should be able to handle that much.

If the current logic provides better experience for you, that can be done in much less code. Just check the stroke time every second or so and set the speed to its inverse. No need for most of the code copied over from the stroker loop.